src/compiler/checker.ts

```diff
@@ -22993,8 +22993,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const sourceObjectFlags = getObjectFlags(source);
             const targetObjectFlags = getObjectFlags(target);
             if (
-                sourceObjectFlags & ObjectFlags.Instantiated && targetObjectFlags & ObjectFlags.Instantiated && source.symbol === target.symbol ||

```

Comment:
yet another question: the comment below says "We have instantiations of the same anonymous type", meaning that if we have two types with the same `symbol`, they're instantiations of the same type (presumably the type represented by the symbol). is that not true then, or is the actual bug that there are two different types with the same symbol?
I know you said in the PR description that "the combo of a type alias and instantiation expression lead to the potential for two of the types to have the same symbol, but have different observed members", but how can that be possible?

Reply:
In this case, they are both instantiations of the same symbol because it's a type alias. But I agree that it's mega confusing, because surely I can invent a type alias whose final type is different depending on its type parameter, and yet we don't crash today. So I think my explanation is wrong and I still need to look at this harder.

Reply:
Like, I feel like it's specifically "type alias pointing to an instantiation expression" that is the problem here, as though there's one extra level deep that was forgotten somewhere, and I just haven't found it.

Or like, this code needs to explicitly exempt instantiation expressions specifically _because_ they're going to have the same symbol, but other things don't. But then, do aliases of aliases not work today?